engine/runtime/src/test/java/org/enso/interpreter/test/LazyGeneratorTest.java

```diff
@@ -0,0 +1,54 @@
+package org.enso.interpreter.test;
+
+import java.net.URI;
+import java.nio.file.Paths;
+import java.util.function.Function;
+import org.enso.polyglot.RuntimeOptions;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Engine;
+import org.graalvm.polyglot.Instrument;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.junit.Test;
+import org.junit.Assert;
+
+public class LazyGeneratorTest {
+  @Test
+  public void evaluation() throws Exception {
+    Engine eng = Engine.newBuilder()
+      .allowExperimentalOptions(true)
+      .option(
+        RuntimeOptions.LANGUAGE_HOME_OVERRIDE,
+        Paths.get("../../distribution/component").toFile().getAbsolutePath()
+      ).build();
+    Context ctx = Context.newBuilder()
+      .engine(eng)
+      .allowIO(true)
+      .build();
+
+    final URI generatorUri = new URI("memory://generator.enso");
+    final Source generatorSrc = Source.newBuilder("enso", """
+    type Generator h t
+
+    natural =
+        gen ~n = if (n >= 100) then this else Generator n (@Tail_Call gen n+1)
+        gen 2
+                                                          
+    Generator.n = case this of
+        Generator n _ -> n
+    Generator.next = case this of
+        Generator _ n -> n                                    
```

Comment:
Why not just put this test in `tests/Test`? Does it in any way rely on the special setup and cannot be run as a regular Enso script?

Reply:
Providing as a regular test as of 5dbbe9c427020bce6771e44c012c1345baa012a4